Add theme styles in the site editor

oandregal opened the pull request:
It looks like we don't enqueue the theme styles in the site editor, but we do in the [widgets](https://github.com/WordPress/gutenberg/blob/e8b901cb7c45fbd23e449b7d0d23da4d7326665c/lib/widgets-page.php#L33-L33) and [post](https://github.com/WordPress/gutenberg/blob/26925dce6c3c5a210dfffe16498a985ed7cbfa3c/lib/client-assets.php#L409-L409) editors. As a result, there're some things that appear broken, such as that a group block with background color doesn't get padding.

This PR enqueues this stylesheet to the site editor as well.

## How to test

- Install and activate this theme: https://github.com/WordPress/gutenberg/files/6041956/emptytheme.zip
- Go to the site editor and add a group block and a paragraph within.
- Set the group's background to some color.

The expected result is that the group block gains some padding.

oandregal committed:
Add theme styles in the site editor

nosolosw commented on lib/full-site-editing/edit-site-page.php:
```diff
@@ -170,6 +170,8 @@ function gutenberg_edit_site_init( $hook ) {
 	wp_enqueue_script( 'wp-format-library' );
 	wp_enqueue_style( 'wp-edit-site' );
 	wp_enqueue_style( 'wp-format-library' );
+	wp_enqueue_style( 'wp-block-library-theme' );
```
I thought we only enqueued these if the theme opted-in into them ([related](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#default-block-styles)), so I'm not sure whether we should load these conditionally.

youknowriad replied:
Yes, I think we should probably load these conditionally. I think @scruffian was working on a related PR lately, not sure if it was for the site editor or post editor.

nosolosw replied:
I think you may be referring to https://github.com/WordPress/gutenberg/pull/29252 Pushed the same logic here: these styles are loaded if the theme declares support for them or if there's no editor styles.

oandregal committed:
Load editor styles conditionally:

- if the theme has declared support
- if the theme doesn't have any editor styles

youknowriad reviewed (approved):
Makes sense to me

scruffian commented on lib/full-site-editing/edit-site-page.php:
```diff
@@ -170,6 +170,14 @@ function gutenberg_edit_site_init( $hook ) {
 	wp_enqueue_style( 'wp-format-library' );
+
+	if (
+		current_theme_supports( 'wp-block-styles' ) ||
```
We have [this same condition](https://github.com/WordPress/gutenberg/blob/9f4f7d41b0129fc934951d0e5b4d2c45a1bf8b5c/lib/client-assets.php#L353) for loading the styles in the post editor. I wonder if we should refactor that to share the code?

youknowriad replied:
If we do refactor this I see a function in `WP_Theme_Config` being introduce here #29667

scruffian reviewed (approved):
LGTM, just one suggestion for improvement.
